src/mmap_sliver.rs

```diff
@@ -0,0 +1,89 @@
+use std::{ io, slice };
+use std::rc::Rc;
+use std::cell::RefCell;
+
+extern crate libc;
+
+use ::Mmap;
+
+#[cfg(target_os = "windows")]
+use super::windows::MmapInner;
+
+#[cfg(not(target_os = "windows"))]
+use super::posix::MmapInner;
+
+pub struct MmapSliver {
+	// The Rc tracks when to drop
+	// The RefCell allows mutable borrows of parent
+	// for flushing
+	parent: Rc< RefCell<Mmap> >,
+	inner: MmapInner
+}
+
+pub unsafe fn carve(mmap: Mmap, ranges: Vec<(usize,usize)> ) -> Vec<MmapSliver> {
```

Comment:
I think this could be simpler with a `split_at_mut`, as you implied in the original post:

```rust
pub fn split_at_mut(mmap: MmapSliver) -> (MmapSliver, MmapSliver)
```

It's the fundamental building block needed to build up a set of ranges specified as a `Vec`.  As a style note, usually you would see a parameter like `ranges` specified as a slice instead of an owned vector, since it's more flexible.